### PR TITLE
remaining review comments from #1613

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

I think this PR covers the remaining comments left in the review of #1613.

The happy path e2e test for channel upgrades now also:
- Prunes acknowledgements after the upgrade.
- Sends an incentivised packet after the upgrade (using a multi message transaction with IBC transfer + pay packet fee). Initially I wrote the test using the async method, but the test didn't pass (packet was not relayed). I tested manually and could reproduce the same behaviour (packets are not relayed if hermes is started after submitting the packet - I have let Luca know about this behaviour). Now the test starts the relayer and then it broadcasts the multi message transaction. Thanks a lot to @chatton for the help debugging the e2e test.

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/core/04-channel/upgrades_test.go [hunk 1]
```diff
@@ -116,6 +137,101 @@ func (s *ChannelTestSuite) TestChannelUpgrade_WithFeeMiddleware_Succeeds() {
...
+		address, err := s.QueryCounterPartyPayee(ctx, chainB, chainBRelayerWallet.FormattedAddress(), channelA.Counterparty.ChannelID)
+		s.Require().NoError(err)
+		s.Require().Equal(chainARelayerWallet.FormattedAddress(), address)
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
+	t.Run("send incentivizes packet", func(t *testing.T) {
```
Comment (damiannolan) [inline e2e/tests/core/04-channel/upgrades_test.go:189]:
```suggestion
	t.Run("send incentivized transfer packet", func(t *testing.T) {
```

File: e2e/tests/core/04-channel/upgrades_test.go [hunk 2]
```diff
@@ -116,6 +137,101 @@ func (s *ChannelTestSuite) TestChannelUpgrade_WithFeeMiddleware_Succeeds() {
...
+		// before adding fees for the packet, there should not be incentivized packets
+		packets, err := s.QueryIncentivizedPacketsForChannel(ctx, chainA, channelA.PortID, channelA.ChannelID)
+		s.Require().NoError(err)
+		s.Require().Empty(packets)
+
+		transferAmount := testvalues.DefaultTransferAmount(chainA.Config().Denom)
+
+		payPacketFeeMsg := feetypes.NewMsgPayPacketFee(testFee, channelA.PortID, channelA.ChannelID, chainAWallet.FormattedAddress(), nil)
+		transferMsg := transfertypes.NewMsgTransfer(channelA.PortID, channelA.ChannelID, transferAmount, chainAWallet.FormattedAddress(), chainBWallet.FormattedAddress(), s.GetTimeoutHeight(ctx, chainB), 0, "")
```
Comment (damiannolan) [inline e2e/tests/core/04-channel/upgrades_test.go:198]: mega nit: normally we write these vars as `msgTransfer` and `msgPayPacketFee` as far as I know

File: e2e/tests/core/04-channel/upgrades_test.go [hunk 3]
```diff
@@ -116,6 +137,101 @@ func (s *ChannelTestSuite) TestChannelUpgrade_WithFeeMiddleware_Succeeds() {
...
+		s.Require().Equal(expected, actualBalance.Int64())
+	})
+
+	t.Run("timeout fee is refunded", func(t *testing.T) {
+		actualBalance, err := s.GetChainANativeBalance(ctx, chainAWallet)
+		s.Require().NoError(err)
+
+		// once the relayer has relayed the packets, the timeout fee should be refunded.
+		// walletA has done to IBC transfers of value testvalues.IBCTransferAmount since the start of the test.
+		expected := testvalues.StartingTokenAmount - 2*testvalues.IBCTransferAmount - testFee.AckFee.AmountOf(chainADenom).Int64() - testFee.RecvFee.AmountOf(chainADenom).Int64()
```
Comment (damiannolan) [inline e2e/tests/core/04-channel/upgrades_test.go:224]: feel free to ignore, maybe this adds more readability 

```suggestion
		expected := testvalues.StartingTokenAmount - (2 * testvalues.IBCTransferAmount) - testFee.AckFee.AmountOf(chainADenom).Int64() - testFee.RecvFee.AmountOf(chainADenom).Int64()
```

File: e2e/tests/core/04-channel/upgrades_test.go [hunk 4]
```diff
@@ -47,15 +48,35 @@ func (s *ChannelTestSuite) TestChannelUpgrade_WithFeeMiddleware_Succeeds() {
...
-		s.AssertTxSuccess(transferTxResp)
+		chainBwalletAmount := ibc.WalletAmount{
+			Address: chainAWallet.FormattedAddress(), // destination address
+			Denom:   chainBDenom,
+			Amount:  sdkmath.NewInt(testvalues.IBCTransferAmount),
+		}
+
+		transferTxResp, err = chainB.SendIBCTransfer(ctx, channelB.ChannelID, chainBWallet.KeyName(), chainBwalletAmount, ibc.TransferOptions{})
+		s.Require().NoError(err)
+		s.Require().NoError(transferTxResp.Validate(), "chain-a ibc transfer tx is invalid")
```
Comment (damiannolan) [inline e2e/tests/core/04-channel/upgrades_test.go:79]:
```suggestion
		s.Require().NoError(transferTxResp.Validate(), "chain-b ibc transfer tx is invalid")
```

**Review by damiannolan — APPROVED** [review]
Looks great! left some small suggestions, LGTM! 🚀

**Review by DimitrisJim — COMMENTED** [review]
small typo and quick Q

File: e2e/tests/core/04-channel/upgrades_test.go (hunk 2 above)
Comment (crodriguezvega) [reply e2e/tests/core/04-channel/upgrades_test.go]: Done. I renamed a couple more in other tests, while I was at it.

File: e2e/testsuite/grpc_query.go [hunk 5]
```diff
@@ -214,6 +214,20 @@ func (s *E2ETestSuite) QueryPacketCommitment(ctx context.Context, chain ibc.Chai
 	return res.Commitment, nil
 }
 
+// QueryPacketAcknowledgements queries the packet acknowledgements on the given chain for the provided channel (optional) list of packet commitment sequences.
```
[the remark this answers is not in the document]
Comment (DimitrisJim) [reply e2e/testsuite/grpc_query.go:217]: ah, I think would be fine w/o it but won't lose sleep over it (i.e I'd assume a list could be empty and that optional refers to the arguments, not the return value).

**Review by DimitrisJim — APPROVED** [review]
lgtm, thanks for going through these! :100:

File: e2e/tests/core/04-channel/upgrades_test.go [hunk 6]
```diff
@@ -47,15 +48,35 @@ func (s *ChannelTestSuite) TestChannelUpgrade_WithFeeMiddleware_Succeeds() {
 	chainBWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
 	chainBAddress := chainBWallet.FormattedAddress()
 
+	chainAwalletAmount := ibc.WalletAmount{
+		Address: chainBWallet.FormattedAddress(), // destination address
```
Comment (colin-axner) [inline e2e/tests/core/04-channel/upgrades_test.go:52]: note: I find this construction to be a little confusing, labeling a wallet as "chainA" but it's address is for chainB?

**Review by colin-axner — APPROVED** [review]
Thanks for following up on these various issues!

File: e2e/tests/core/04-channel/upgrades_test.go (hunk 6 above)
Comment (crodriguezvega) [reply e2e/tests/core/04-channel/upgrades_test.go]: I renamed it!

**Pull request merged**